corehq/motech/tests/test_rate_limiter.py

```diff
@@ -0,0 +1,73 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+from corehq.motech.rate_limiter import _rate_limit_repeater
+from corehq.util.test_utils import flag_enabled
+
+
+@patch('corehq.motech.rate_limiter.metrics_counter')
+@patch('corehq.motech.rate_limiter.repeater_rate_limiter.allow_usage')
+@patch('corehq.motech.rate_limiter.repeater_attempts_rate_limiter.allow_usage')
+@patch('corehq.motech.rate_limiter.global_repeater_rate_limiter.allow_usage')
+class TestRateLimitRepeater(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.domain = 'test-domain'
+        cls.repeater_id = 'test-repeater-id'
+
+    def test_rate_limited(self, global_allowed, repeater_attempts_allowed, repeater_allowed, metrics_counter):
```

Comment:
Is this test doing anything not covered by `test_repeater_rate_limited`?

Reply:
Hey @jingcheng16 

This is doing a check when rate limiting is being applied for all factors, so all possible scenarios where repeater would be allowed to go through are set to `False`.
`repeater_attempts_allowed` isn't applicable here since it is only relevant when there is no global rate limiting.

`test_repeater_rate_limited` is checking for rate limiting conditions specifically for rate limiting at the repeater level.